src/rust/engine/process_execution/src/docker.rs

```diff
@@ -338,29 +334,59 @@ impl<'a> super::CommandRunner for CommandRunner<'a> {
           self.keep_sandboxes,
         )?;
 
+        // Obtain ID of the base container in which to run the execution for this process.
+        let (container_id, named_caches) = {
+          let ProcessExecutionStrategy::Docker(image) = &req.execution_strategy else {
+            return Err(ProcessError::Unclassified(
+                "The Docker execution strategy was not set on the Process, but \
+                 the Docker CommandRunner was used.".to_owned()))
+          };
+
+          self
+            .container_cache
+            .container_for_image(image, &req.platform, &context.build_id)
+            .await?
+        };
+
         // Start working on a mutable version of the process.
         let mut req = req;
 
-        // Update env, replacing `{chroot}` placeholders with the path to the sandbox
-        // within the Docker container.
-        let sandbox_relpath = workdir
-          .path()
-          .strip_prefix(&self.work_dir_base)
-          .map_err(|err| {
-            format!("Internal error - base directory was not prefix of sandbox directory: {err}")
-          })?;
-        let sandbox_path_in_container = Path::new(&SANDBOX_BASE_PATH_IN_CONTAINER)
-          .join(sandbox_relpath)
-          .into_os_string()
-          .into_string()
-          .map_err(|s| {
-            format!("Unable to convert sandbox path to string due to non UTF-8 characters: {s:?}")
-          })?;
-        apply_chroot(&sandbox_path_in_container, &mut req);
-        log::trace!(
-          "sandbox_path_in_container = {:?}",
-          &sandbox_path_in_container
-        );
+        // Compute the absolute working directory within the container, and update the env to
+        // replace `{chroot}` placeholders with the path to the sandbox within the Docker container.
+        let working_dir = {
+          let sandbox_relpath =
+            workdir
+              .path()
+              .strip_prefix(&self.work_dir_base)
+              .map_err(|err| {
+                format!(
+                  "Internal error - base directory was not prefix of sandbox directory: {err}"
+                )
+              })?;
+          let sandbox_path_in_container = Path::new(&SANDBOX_BASE_PATH_IN_CONTAINER)
+            .join(sandbox_relpath)
+            .into_os_string()
+            .into_string()
+            .map_err(|s| {
+              format!("Unable to convert sandbox path to string due to non UTF-8 characters: {s:?}")
+            })?;
+          apply_chroot(&sandbox_path_in_container, &mut req);
+          log::trace!(
+            "sandbox_path_in_container = {:?}",
+            &sandbox_path_in_container
+          );
+
+          req
+            .working_directory
+            .as_ref()
+            .map(|relpath| Path::new(&sandbox_path_in_container).join(relpath))
```

Comment:
In this closure and the next the wrapping Path::new appears to be un-needed.

Reply:
As factored, those are necessary because `sandbox_path_in_container` is a string (since it was used for a find-replace a little bit above this), and both of the closures do more than just the wrapping.